operator/internal/controller/authorizationmodel/authorizationmodel_controller_test.go

```diff
@@ -0,0 +1,266 @@
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Image: getLowercaseUUID(),
+							Name:  getLowercaseUUID(),
+						},
+						{
+							Image: getLowercaseUUID(),
+							Name:  getLowercaseUUID(),
+							Env: []corev1.EnvVar{
+								{Name: extensionsv1.OpenFgaAuthModelIdEnv, Value: getLowercaseUUID()},
+							},
+						},
+						{
+							Image: getLowercaseUUID(),
+							Name:  getLowercaseUUID(),
+							Env: []corev1.EnvVar{
+								{Name: extensionsv1.OpenFgaStoreIdEnv, Value: getLowercaseUUID()},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func validateDeployment(deploymentName, namespaceName, storeId, authModelId string, modelVersion extensionsv1.ModelVersion) {
+	Eventually(func() error {
+		deployment := &appsV1.Deployment{}
+		if err := k8sClient.Get(ctx, types.NamespacedName{
+			Name:      deploymentName,
+			Namespace: namespaceName,
+		}, deployment); err != nil {
+			return err
+		}
+
+		// Validate that all containers have the necessary environment variables
+		for _, container := range deployment.Spec.Template.Spec.Containers {
+			foundStoreId := false
+			foundAuthModelId := false
+
+			// Check environment variables in each container
+			for _, envVar := range container.Env {
+				if envVar.Name == extensionsv1.OpenFgaStoreIdEnv && envVar.Value == storeId {
+					foundStoreId = true
+				}
+				if envVar.Name == extensionsv1.OpenFgaAuthModelIdEnv && envVar.Value == authModelId {
+					foundAuthModelId = true
+				}
+			}
+
+			if !foundStoreId {
+				return fmt.Errorf("container %s does not have env var %s with value %s", container.Name, extensionsv1.OpenFgaStoreIdEnv, storeId)
+			}
+			if !foundAuthModelId {
+				return fmt.Errorf("container %s does not have env var %s with value %s", container.Name, extensionsv1.OpenFgaAuthModelIdEnv, authModelId)
+			}
+		}
+
+		// Validate that the deployment has the annotation OpenFgaAuthModel with the correct version
+		if deployment.Annotations[extensionsv1.OpenFgaAuthModelVersionLabel] != modelVersion.String() {
+			return fmt.Errorf("deployment does not have annotation %s with value %s", extensionsv1.OpenFgaAuthModelVersionLabel, modelVersion.String())
+		}
+		if deployment.Annotations[extensionsv1.OpenFgaStoreIdUpdatedAtAnnotation] != MockTimeAsString() {
+			return fmt.Errorf("deployment does not have annotation %s with value %s", extensionsv1.OpenFgaStoreIdUpdatedAtAnnotation, MockTimeAsString())
+		}
+		if deployment.Annotations[extensionsv1.OpenFgaAuthIdUpdatedAtAnnotation] != MockTimeAsString() {
+			return fmt.Errorf("deployment does not have annotation %s with value %s", extensionsv1.OpenFgaAuthIdUpdatedAtAnnotation, MockTimeAsString())
+		}
+
+		return nil
+	}, duration, interval).Should(Succeed())
+}
+
+func validateNoEventsFound(events <-chan string) {
+	Consistently(func() bool {
+		select {
+		case <-events:
```

Comment:
what if this is empty response because channel was closed?

Reply:
That's a valid point.

I wouldn't expect the channel to close until after the test is complete. Therefore, if the channel closes prematurely and causes the test to fail, it indicates unexpected behavior in either case.